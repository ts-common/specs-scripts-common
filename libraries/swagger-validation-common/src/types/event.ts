export type PipelineResult = "Success" | "Failure" | "TimedOut";

export type PipelineRun = {
  source: "GitHub" | "OpenAPIHub"
  unifiedPipelineTaskKey: string; // a unified pipeline task key, e.g. LintDiff, Semantic
  unifiedPipelineBuildId: string; // a unique build id unified pipeline assigned for each completed pipeline build id
  pipelineBuildId: string; // the id of the record for the completed azure pipeline build.
  pipelineJobId: string; // the id of the record for the completed azure pipeline job.
  pipelineTaskId: string; // the id of the record for the completed azure pipeline task.
};

export type PipelineStatus = "InProgress" | "Completed";

export type InProgressEvent = PipelineRun & {
  status: "InProgress";
};

export type CompletedEvent = PipelineRun & {
  status: "Completed";
  result: PipelineResult;
  logPath: string;
};

export type PipelineEvent = InProgressEvent | CompletedEvent;
