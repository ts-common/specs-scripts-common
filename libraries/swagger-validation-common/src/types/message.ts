export type MessageLevel = "Info" | "Warning" | "Error";

export type JsonPath = {
  tag: string; // meta info about the path, e.g. "swagger" or "example"
  path: string;
};

export type MesssageContext = {
  toolVersion: string;
};

export type Extra = {
  [key: string]: any;
};

export type BaseMessageRecord = {
  level: MessageLevel;
  message: string;
  time: Date;
  context?: MesssageContext;
  extra?: Extra;
};

export type ResultMessageRecord = BaseMessageRecord & {
  type: "Result";
  id?: string;
  code?: string;
  docUrl?: string;
  paths: JsonPath[];
};

export type RawMessageRecord = BaseMessageRecord & {
  type: "Raw";
};

export type MarkdownMessageRecord = BaseMessageRecord & {
  type: "Markdown";
  mode: "replace" | "append";
};

export type MessageRecord = ResultMessageRecord | RawMessageRecord | MarkdownMessageRecord;

export type MessageLine = MessageRecord | MessageRecord[];
