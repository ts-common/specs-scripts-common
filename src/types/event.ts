type Stage = "Queued" | "InProgress" | "Completed";

type Result = "Success" | "Failure" | "TimedOut";

export type PipelineRun = {
  buildId: string;
  pipelineBuildId: string;
  pipelineTaskId: string;
};

export type InProgressTask = PipelineRun & {
  stage: "InProgress";
};

export type CompletedTask = PipelineRun & {
  stage: "Completed";
  result: Result;
  logPath: string;
};
