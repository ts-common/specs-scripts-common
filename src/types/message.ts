export type MessageLevel = "Info" | "Warning" | "Error";

export type JsonPath = {
  tag: string; // meta info about the path, e.g. "swagger" or "example"
  path: string;
};

export type Extra = {
  [key: string]: any;
};

export type BaseMessageRecord = {
  level: MessageLevel;
  message: string;
  time: Date;
  extra?: Extra;
};

export type ResultMessageRecord = BaseMessageRecord & {
  id?: string;
  code?: string;
  docUrl?: string;
  paths: JsonPath[];
};

export type RawMessageRecord = BaseMessageRecord;

export type RawMessage = {
  type: "Raw";
  data: RawMessageRecord[];
};

export type ResultMessage = {
  type: "Result";
  data: ResultMessageRecord[];
};

export type Message = ResultMessage | RawMessage;
