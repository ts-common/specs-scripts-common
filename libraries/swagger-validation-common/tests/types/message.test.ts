import { MessageRecord } from "../../src/types/message";
import rawMessage from "./rawMessage.json";
import markdownMessage from "./markdownMessage.json";
import resultMessage from "./resultMessage.json";

describe("Message", () => {
  test("should parse raw message", () => {
    const rawMsg = JSON.stringify(rawMessage);
    const msg: MessageRecord = JSON.parse(rawMsg);
    expect(msg.type).toBe("Raw");
    expect(msg).toMatchSnapshot();
  });

  test("should parse result message", () => {
    const resultMsg = JSON.stringify(resultMessage);
    const msg: MessageRecord = JSON.parse(resultMsg);
    expect(msg.type).toBe("Result");
    expect(msg).toMatchSnapshot();
  });

  test("should parse markdown message", () => {
    const markdownMsg = JSON.stringify(markdownMessage);
    const msg: MessageRecord = JSON.parse(markdownMsg);
    expect(msg.type).toBe("Markdown");
    expect(msg).toMatchSnapshot();
  });
});
