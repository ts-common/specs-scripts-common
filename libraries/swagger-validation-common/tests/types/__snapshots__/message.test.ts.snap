// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message should parse raw message 1`] = `
Object {
  "extra": Object {
    "role": "sdk automation",
  },
  "level": "Error",
  "message": "the tool failed",
  "time": "2012-04-23T18:25:43.511Z",
  "type": "Raw",
}
`;

exports[`Message should parse result message 1`] = `
Object {
  "code": "InvalidVerbUsed",
  "docUrl": "https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/openapi-authoring-automated-guidelines.md#r2044",
  "extra": Object {
    "otherDetails": Object {
      "test": "hello",
    },
    "readmeUrl": "http://readme.example.url",
  },
  "id": "R2044",
  "level": "Info",
  "message": "Each operation definition must have a HTTP verb and it must be DELETE/GET/PUT/PATCH/HEAD/OPTIONS/POST/TRACE.",
  "paths": Array [
    Object {
      "path": "specification/databox/resource-manager/Microsoft.DataBox/stable/2019-09-01/databox.json:123:12",
      "tag": "swagger",
    },
    Object {
      "path": "specification/databox/resource-manager/Microsoft.DataBox/stable/2019-09-01/databox.json:126:13",
      "tag": "example",
    },
  ],
  "time": "2012-04-23T18:25:43.511Z",
  "type": "Result",
}
`;
